src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectSystem/RazorProjectHostBase.cs

```diff
@@ -184,6 +184,11 @@ protected void UpdateProjectUnsafe(HostProject project)
             }
             else if (Current == null && project != null)
             {
+                // Just in case we somehow got in a state where VS didn't tell us that solution close was finished, lets just
+                // ensure we're going to actually do something with the new project that we've just been told about.
+                // If VS did tell us, then this is a no-op.
+                projectManager.SolutionOpened();
```

Comment:
Ya I feel your pain here 😢. Should we just rely on this method for detecting solution closed in our own way? Aka look at how many projects are known and mark solution open / closed based on that? Only thinking about using this as the primary source of truth because it feels a little rough to have two sources of truth for "solution is closed / opened"

Reply:
I don't think we should be trying to detect _close_ solution, but rely on the event for that. For open, I could see an argument to get rid of the event (because its _after_ solution open, not before), but I'm personally fine with having multiple signals, given that its just a bit flip.